oracle-price-monitor/src/autotask.js

```diff
@@ -0,0 +1,270 @@
+    cTokenAddress,
+    CTOKEN_ABI,
+    provider,
+  );
+  const underlyingTokenAddress = await cTokenContract.underlying();
+
+  let decimals;
+  let symbol;
+  if (oddTokens.indexOf(underlyingTokenAddress.toLowerCase()) !== -1) {
+    const underlyingTokenContract = new ethers.Contract(
+      underlyingTokenAddress,
+      MAKER_TOKEN_ABI,
+      provider,
+    );
+
+    decimals = await underlyingTokenContract.decimals();
+    // need to convert decimals from uint256 to uint8
+    decimals = parseInt(decimals.toString(), 10);
+
+    symbol = await underlyingTokenContract.symbol();
+    // need to convert symbol from bytes32 to string
+    symbol = ethers.utils.parseBytes32String(symbol);
+  } else {
+    const underlyingTokenContract = new ethers.Contract(
+      underlyingTokenAddress,
+      TOKEN_ABI,
+      provider,
+    );
+    decimals = await underlyingTokenContract.decimals();
+    symbol = await underlyingTokenContract.symbol();
+  }
+  return { decimals, symbol };
+}
+
+function formatAmountString(amount, decimals) {
+  console.log("amount here", amount)
+  const amountBN = ethers.BigNumber.from(amount);
+  const divisorBN = ethers.BigNumber.from(10).pow(decimals);
+
+  // the ethers BigNumber implementation will discard the decimal
+  // portion of the value when we perform the division
+  let resultString = amountBN.toString();
+  if (resultString.length <= decimals) {
+    resultString = `0.${'0'.repeat(decimals - resultString.length)}${resultString[0]}`;
+  } else {
+    resultString = amountBN.div(divisorBN).toString();
+  }
+
+  // format the number to have comma separators for powers of 1000
+  const internationalNumberFormat = new Intl.NumberFormat('en-US');
+  return internationalNumberFormat.format(resultString);
+}
+
+async function createDiscordMessage(reporterPrice, cTokenAddress, transactionHash, provider) {
+  const { decimals, symbol } = await getDecimalsAndSymbol(cTokenAddress, provider);
+
+  const amountString = formatAmountString(reporterPrice, decimals);
+
+  console.log("amount string here", amountString)
+
+  // // construct the Etherscan transaction link
+  const etherscanLink = `[TX](<https://etherscan.io/tx/${transactionHash}>)`;
+
+  return `${etherscanLink} 🚫 reported price of **${amountString}** for **${symbol}** was rejected`;
+}
+
+// post to discord
+async function postToDiscord(url, message) {
+  let response;
+  try {
+    // perform the POST request
+    response = needle.post(url, {content: message}, {json: true} )
+  } catch (error) {
+    // is this a "too many requests" error (HTTP status 429)
+    if (error.response && error.response.status === 429) {
+      // the request was made and a response was received
+      // try again after waiting 5 seconds
+      // eslint-disable-next-line no-promise-executor-return
+      const promise = new Promise((resolve) => setTimeout(resolve, 5000));
+      await promise;
+      response = needle.post(url, {content: message}, {json: true} )
```

Comment:
Replace `needle.post` with `axios`. It's unnecessary to have to modules that do the same thing and we use `axios` everywhere else.